route_guide/route_guide_client.cc

```diff
@@ -23,238 +23,347 @@
+  // comparable to `GetFeature()`, but instead of sending a single `Point`
+  // the client sends a `Rectangle`. The server responds with a stream of
+  // messages containing addresses that fall inside the Rectangle's area.
+  auto ListFeatures() {
+    struct Data {
+      Rectangle rectangle; // Request from the client.
+      Feature feature; // Response from the server.
+    };
+    return client_->Call<RouteGuide, Rectangle, Stream<Feature>>(
+               "ListFeatures")
+        | Client::Handler()
+              .context(Data{Rectangle{}, Feature{}})
+              // Prepare the rectangle area and send to the server as a
+              // request.
+              .ready([](auto&& data, auto& call) {
+                data.rectangle.mutable_lo()->set_latitude(400000000);
+                data.rectangle.mutable_lo()->set_longitude(-750000000);
+                data.rectangle.mutable_hi()->set_latitude(420000000);
+                data.rectangle.mutable_hi()->set_longitude(-730000000);
+                std::cout << "Looking for features between 40, -75"
+                          << " and 42, -73" << std::endl;
+                call.WriteLast(data.rectangle);
+              })
+              // The .body() callback is invoked every time the server
+              // sends a response. The response contains the addresses
+              // that fall within the requested rectangle's area.
+              .body([this](auto&& data, auto& call, auto&& feature) {
+                // Check if the response from the server is valid.
+                if (!feature) {
+                  return;
+                }
+                std::cout
+                    << "Found feature called " << feature->name()
+                    << " at "
+                    << feature->location().latitude() / kCoordFactor_
+                    << ", "
+                    << feature->location().longitude() / kCoordFactor_
+                    << std::endl;
+              })
+              // .finished callback is invoked when the call is finished.
+              // In this callback we just start our continuation.
+              .finished([](
+                            auto&& data,
+                            auto& continuation,
+                            auto&& response) {
+                continuation.Start(grpc::Status::OK);
+              });
   }
 
-  void RecordRoute() {
-    Point point;
-    RouteSummary stats;
+  // RecordRoute() function implements the client streaming RPC. The client
+  // sends 10 Point's messages to the server. Each message consists of the
+  // Point's data stucture. The server responds with a unique message which
+  // contains the number of points, number of feautures found in the db,
+  // the resulting distance calculated by using all Point's coordinates and
+  // the resulting time.
+  auto RecordRoute() {
+    struct Data {
+      size_t kPoints; // The number of points to send to the server.
```

Comment:
How about this slightly clearer name (using underscores as is our style):
```suggestion
      size_t num_points; // The number of points to send to the server.
```